### ISSUE TITLE
[Bug]: Errors Thrown During a Playwright Test Aren't Shown in the (Interactive Test Runner) Errors Tab

### ISSUE THREAD

**Issue by jwalkerinterpres** [microsoft/playwright#29622]
### Version

1.42.0-alpha-1707816346000

### Steps to reproduce

1. Create a test with a `beforeAll`
2. Have that `beforeAll` call a helper function
3. Throw an explicit error in that function (`throw new Error('foo')`);
4. The Errors tab shows no errors (and neither do the Logs or Console tabs):
<img width="1253" alt="image" src="https://github.com/microsoft/playwright/assets/152227747/2536016a-db1b-48fe-bbd5-050033a961f2">

### Expected behavior

All errors thrown, no matter where or how, should be reported.  This applies _a hundred fold_ when those errors are stopping test execution.

### Actual behavior

Tests fail silently!

### Additional context

I will happily provide a repro example as soon as https://github.com/microsoft/playwright/issues/29620 is resolved.

### Environment

```shell
System:
    OS: macOS 14.3.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 65.91 MB / 32.00 GB
  Binaries:
    Node: 21.2.0 - /usr/local/bin/node
    Yarn: 1.22.21 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
  IDEs:
    VSCode: 1.86.2 - /usr/local/bin/code
  Languages:
    Bash: 3.2.57 - /bin/bash

P.S. Why not just make this display the Playwright version? 
You could then simplify this bug report template by eliminating the initial "Version" field.
```

**Comment by yury-s** [comment]
Cannot reproduce it with https://github.com/yury-s/bug-29622, you can clone that repository to provide your reproduction case.

**Comment by yury-s** [comment]
I see what you mean, in the UI mode the error is not in the list indeed.

**Comment by dgozman** [comment]
This has been fixed recently, you can try it out with `npm install @playwright/test@1.43.0-alpha-2024-03-18`. Will be released in the next v1.43 release.
